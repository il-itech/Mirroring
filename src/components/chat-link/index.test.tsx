import { shallow, ShallowWrapper } from 'enzyme';

import { ChatLink } from '.';

const handleToggleDrawer = jest.fn();

const props = {
  href: '/',
  firstname: 'Zakk',
  lastname: 'Wylde',
  lastMessage: 'How that going ?',
  avatar: 'avatarHref',
  handleToggleDrawer,
};

describe('<ChatLink />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    handleToggleDrawer.mockClear();
    wrapper = shallow(<ChatLink {...props} />);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('handleToggleDrawer should be called', () => {
    wrapper.find({ 'data-test-id': 'Button' }).simulate('click');

    expect(props.handleToggleDrawer).toHaveBeenCalled();
  });
});
