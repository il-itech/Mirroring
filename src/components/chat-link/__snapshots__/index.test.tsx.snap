// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatLink /> should match snapshot 1`] = `
<Link
  href="/"
>
  <WithStyles(ForwardRef(Button))
    classes={
      Object {
        "root": "d-flex justify-content-start w-100 py-1_5 px-2 text-white text-capitalize link-hover",
      }
    }
    data-test-id="Button"
    onClick={[MockFunction]}
  >
    <WithStyles(ForwardRef(Avatar))
      src="avatarHref"
    />
    <div
      className="d-flex flex-column align-items-start ml-2 text-truncate"
    >
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        Zakk
         
        Wylde
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="text-left text-mischka text-truncate"
        variant="body2"
      >
        How that going ?
      </WithStyles(ForwardRef(Typography))>
    </div>
  </WithStyles(ForwardRef(Button))>
</Link>
`;
